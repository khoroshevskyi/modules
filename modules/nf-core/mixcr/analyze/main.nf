process MIXCR_ANALYZE {
    tag "$meta.id"
    label 'process_medium'

    secret 'MI_LICENSE'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7dccca2db544d708a462b2ba49117b41e32141d8eb46f4dbc8730e210d2cdbae/data':
        'community.wave.seqera.io/library/mixcr:4.7.0--bb57944ca92aeb74' }"  //ghcr.io/milaboratory/mixcr/mixcr:4.7.0

    containerOptions "${ workflow.containerEngine == 'singularity' ?
        '-B \$HOME' :
        '' }"

    input:
    tuple val(meta), path(reads)
    val preset
    val species

    output:
    tuple val(meta), path("*clones*.tsv"), emit: clones
    tuple val(meta), path("*.txt"),        emit: reports
    tuple val(meta), path("*.clns"),       emit: clns, optional: true
    tuple val(meta), path("*.vdjca"),      emit: vdjca, optional: true
    path "versions.yml",                   emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def java_mem = task.memory ? "-Xmx${task.memory.toGiga()}g" : ''
    """
    mixcr $java_mem analyze \\
        $preset \\
        --species $species \\
        $args \\
        --threads $task.cpus \\
        ${reads} \\
        $prefix

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mixcr: \$(mixcr -v 2>&1 | sed -n '1p' | sed -E 's/MiXCR v([0-9\\.]+).*/\1/' || true)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mixcr -v 
    touch ${prefix}.clones_TRA.tsv
    touch ${prefix}.clones_TRB.tsv
    touch ${prefix}.clns
    touch ${prefix}.vdjca
    touch ${prefix}.report.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mixcr: \$(mixcr -v 2>&1 | sed -n '1p' | sed -E 's/MiXCR v([0-9\\.]+).*/\1/' || true)
    END_VERSIONS
    """
}
