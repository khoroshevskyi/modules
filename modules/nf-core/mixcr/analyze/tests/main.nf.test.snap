{
    "10xgenomics vdj - fastqs - stub": {
        "content": [
            [
                "versions.yml:md5,6e7142daa5212cb45dbee665f713f700"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-09T09:44:14.489185"
    },
    "10xgenomics vdj - fastq": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.clones.tsv:md5,bdf3ad0dfd9b3fee124bfa80e268e534"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.extended.vdjca:md5,5ac4a6d182ab48c670154fbbc656c364"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.contigs.clns:md5,ffb386656fc9d45f396ea8cd30e1dfef"
                ]
            ],
            [
                "versions.yml:md5,6e7142daa5212cb45dbee665f713f700"
            ],
            {
                "MIXCR_ANALYZE": {
                    "mixcr": "4.7.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-09T09:51:02.542818"
    }
}